src/main/java/eu/greev/dcbot/ticketsystem/TicketListener.java

```diff
@@ -38,6 +37,15 @@ public class TicketListener extends ListenerAdapter {
     private final TicketService ticketService;
     private final Config config;
 
+    @Override
+    public void onChannelUpdateArchived(ChannelUpdateArchivedEvent event) {
+        if (ticketService.getTicketByChannelId(event.getChannel().asThreadChannel().getParentMessageChannel().getIdLong()) == null
+                || !Boolean.TRUE.equals(event.getNewValue())) {
+            return;
+        }
+        ((ThreadChannel) event.getChannel()).getManager().setArchived(false).queue();
```

Comment:
Does it make sense here to do something like if (event.getChannel() instance of ThreadChannel channel) {}?

Reply:
No since the ChannelUpdateArchivedEvent is limited to ThreadChannel, so the channel can't be something else

Reply:
What about future Discord Updates?